pandas/core/indexes/base.py

```diff
@@ -2019,15 +2019,35 @@ def __contains__(self, key):
             return False
 
     _index_shared_docs['contains'] = """
-        return a boolean if this key is IN the index
+        Return a boolean indicating whether this key is in the index.
 
         Parameters
         ----------
-        key : object
+        key : label
+            The key can be of the same type as the label of :class:`Index`,
+            hence immutable-like and 1-dimensional if it is a tuple.
 
         Returns
         -------
-        boolean
+        bool
+            Result indicating whether the key search is in the index.
+
+        See Also
+        --------
+        Index.isin : Returns an ndarray of boolean dtype indicating whether the
+            list-like key is in the index.
+
+        Examples
+        --------
+        >>> idx = pd.Index([1, 2, (3, 4), 5])
```

Comment:
I would prefer an example with a simple values like ``pd.Index(list('abcd'))``. This is a rather esoteric case.